octoprint_mrbeam/analytics/usage_handler.py

```diff
@@ -880,3 +877,307 @@ def _migrate_old_airfilter_structure_to_new(self):
                 prefilter, carbon_filter, serial
             )
         )
+
+    def get_carbon_filter_usage(self):
+        """
+        Calculate the usage of the carbon filter in percent.
+
+        Returns:
+            The usage of the carbon filter in percent.
+        """
+        if self._airfilter is None:
+            return -1
+        if self._airfilter.model_id in AirFilter.AIRFILTER3_MODELS:
+            percent = self._calculate_af3_filter_usage(
+                filter_stage=AirFilter.CARBONFILTER
+            )
+        else:
+            percent = self._calculate_af2_filter_usage(
+                filter_stage=AirFilter.CARBONFILTER
+            )
+        return round(percent)
+
+    def get_prefilter_usage(self):
+        """
+        Calculate the usage of the prefilter in percent.
+
+        Returns:
+            The usage of the prefilter in percent.
+        """
+        if self._airfilter is None:
+            return -1
+        if self._airfilter.model_id in AirFilter.AIRFILTER3_MODELS:
+            percent = self._calculate_af3_filter_usage(filter_stage=AirFilter.PREFILTER)
+        else:
+            percent = self._calculate_af2_filter_usage(filter_stage=AirFilter.PREFILTER)
+        return round(percent)
+
+    def _calculate_af2_filter_usage(self, filter_stage):
+        """
+        Calculate the usage of the given filter stage in percent.
+
+        Args:
+            filter_stage: The filter stage to calculate the usage for.
+
+        Returns:
+            The usage of the given filter stage in percent.
+        """
+        # get the correct data for the filter stage
+        if filter_stage == AirFilter.PREFILTER:
+            stage_usage_time = self._get_prefilter_usage_time()
+        elif filter_stage == AirFilter.CARBONFILTER:
+            stage_usage_time = self._get_carbon_filter_usage_time()
+
+        usage_time = stage_usage_time if stage_usage_time is not None else 0
```

Comment:
Another option (just a suggestion)
```suggestion
        usage_time = stage_usage_time or 0
```